src/games/strategy/thread/LockUtil.java

```diff
@@ -21,42 +24,54 @@
  * you are considering your ambitious multi-threaded code a mistake, and you are trying to limit the damage.
  * <p>
  */
-public class LockUtil {
+public final class LockUtil {
+  private static final LockUtil instance = new LockUtil();
```

Comment:
You know, I must have read Bloch cover-to-cover at least three times, and I don't remember this gem (emphasis Bloch's):

> While this approach has yet to be widely adopted, **a single-element enum type is the best way to implement a singleton**.

Thanks for the nudge!